FoodTracker/Controllers/MealViewController.swift

```diff
@@ -0,0 +1,119 @@
+//
+//  MealViewController.swift
+//  FoodTracker
+//
+//  Created by BrainX IOS 4 on 2020-12-14.
+//
+import os.log
+import UIKit
+
+class MealViewController: UIViewController {
+
+    //MARK: Outlets
+    @IBOutlet weak var mealName: UITextField!
+    @IBOutlet weak var mealImage: UIImageView!
+    @IBOutlet weak var ratingControl: RatingControl!
+    @IBOutlet weak var saveButton: UIBarButtonItem!
+    
+    //MARK: Properties
+    var meal: Meal?
+    
+    //MARK: ViewController Lifecycle Methods
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        mealName.delegate = self
+        setMeal()
+        updateSavedButtonStatus()
+    }
+    
+    //MARK: Actions
+    @IBAction
+    func selectImageFromGallery(_ sender: UITapGestureRecognizer) {
+        mealName.resignFirstResponder()
+        let imagePickerController = UIImagePickerController()
+        imagePickerController.sourceType = .photoLibrary
+        imagePickerController.delegate = self
+        present(imagePickerController, animated: true, completion: nil)
+    }
+    
+    //MARK: Private Methods
+    private func updateSavedButtonStatus() {
+        let textField = mealName.text ?? ""
+        saveButton.isEnabled = !textField.isEmpty
+    }
+    
+    private func setMeal() {
+        if let meal = meal {
```

Comment:
Guard would be more suitable there because:
1) You don't want to do anything if there is no meal, so guard can be used to early exit.
2) With guard you can get rid of nested blocks.